### PR TITLE
update README: Adding ToC, how to use the notebooks, descriptions.  Little cleanup

### PR DESCRIPTION
**Pull request by Urfoex**
epic &4
issue #35

### CODE REVIEW

**Comment by liuzicheng1987** [comment]
Why are we using podman here? I think Docker is the standard here and using a tool that many people have never heard of will make adoption harder.

**Comment by Urfoex** [comment]
> Why are we using podman here? I think Docker is the standard here and using a tool that many people have never heard of will make adoption harder.

Podman is a mostly hazel-free drop-in replacement for Docker. Because of its higher security, it is included in Redhat since version RHEL 8:

https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html/8.0_release_notes/notable_changes_to_containers

https://www.heise.de/news/Red-Hat-Enterprise-Linux-8-freigeben-Mehr-Flexibilitaet-bei-Software-Versionen-4416725.html#nav_container_im__3

Not to mention: companies with more then 250 people and Mac and Windows users probably evaluated alternatives like Podman-Desktop and [Rancher-Desktop](https://rancherdesktop.io/) because of:
"Commercial use of Docker Desktop at a company of more than 250 employees OR more than $10 million in annual revenue requires a paid subscription (Pro, Team, or Business)."  (https://www.docker.com/products/docker-desktop/)

Define "many people". 
https://github.com/containers/podman has 21K Stars. 

While the docker open source part seems to be now moby
https://stackoverflow.com/questions/47793812/where-is-the-docker-source-code-located-github-repo
https://github.com/moby/moby
with 67K stars.

"Docker being the standard" for what?
The container world is much bigger. 
https://en.wikipedia.org/wiki/List_of_cluster_management_software
https://en.wikipedia.org/wiki/OS-level_virtualization

For simple users trying the notebooks it would be nicer to have a full ready image on a registry.
It seems, there is a getml account;
https://hub.docker.com/u/getml
It would be nicer, to have a full image with notebook and engine on their and ready to run, instead of having the user to build it.
Or on another registry, like 

-     quay.io (redhat)
-     gcr.io (google)
-     public.ecr.aws (amazon)
-     ghcr.io (github)
-     ...

Or simply download the notebooks and run them locally inside their Jupyer-Lab.

Not to mention, podman is a drop-in replacement for docker.
In the repository, there is still just Dockerfile and docker-compose.yml. Just making sure, that the software and libraries inside are provided and run-able as user and not as root.
There is no extra adoption needed. Just reaching a wider audience.

**Comment by srnnkls** [comment]
Thank you @Urfoex. While I agree with everything you said, I would argue many people – in particular in the data science community – are not familiar with the nuances of Linux containerization. As docker is the best known implementation, we should use that throughout the examples. If you like, maybe, you can reference podman in the readme.

**Comment by Urfoex** [comment]
Oh, in the docker/README.md. Sorry, I missed that part.
Adjusted.

File: docker/README.md [hunk 1]
```diff
@@ -4,40 +4,48 @@
...
 
 Look for this line in the output and copy-paste it in your browser:
 
-> Or copy and paste one of these URLs:
->
-> http://localhost:8888/lab?token=xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
+```
+Or copy and paste one of these URLs:
+
+http://localhost:8888/lab?token=xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
```
Comment (alxn4) [inline docker/README.md:40]: Please find a better placeholder for `xxxxxx`. Candidate: <randomly_generated_token>

File: docker/README.md [hunk 2]
```diff
@@ -4,40 +4,48 @@
...
 
 Look for this line in the output and copy-paste it in your browser:
 
-> Please open a browser and point it to the following URL:
->
-> http://localhost:1709/#/token/xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
+```
+Please open a browser and point it to the following URL:
+
+http://localhost:1709/#/token/xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
```
Comment (alxn4) [inline docker/README.md:50]: same here

File: README.md [hunk 3]
```diff
@@ -18,93 +16,681 @@
...
+
+> [!TIP]  
+> Install the [trail of the enterprise version](https://www.getml.com/download) via the  [Install getML on Linux guide](https://docs.getml.com/latest/home/installation/linux.html#install-getml-on-linux) to try the enterprise features.
+
+With the last command, Jupyter-Lab should automatically open in the browser. If not, look for the following lines in the output and copy-paste it in your browser:
+
+```
+Or copy and paste one of these URLs:
+
+http://localhost:8888/lab?token=xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
```
Comment (alxn4) [inline README.md:113]: Please find a better placeholder for `xxxxxx...`. Candidate: `<randomly_generated_token>`

**Review by alxn4 — CHANGES_REQUESTED** [review]

File: README.md [hunk 4]
```diff
@@ -18,93 +16,681 @@
...
+
+## Reading Online 
+
+As github renders the notebooks, they can each be viewed by just opening and scrolling through them. For convenience, the output of each cells execution is included.
+
+## Experimenting Locally
+
+For testing the notebooks completely, having them run through, playing with parameters, they should be run on a local machine. While Windows and Mac users just have the containerized option, Linux users can also choose a local one.
+
+### Using Docker & Podman
```
[the remark this answers is not in the document]
Comment (srnnkls) [reply README.md:54]: I would propose just using a arch agnostic base image, e.g. the official debian-based `python:3.11` image. I don’t see any benefits in using the outdated manylknux2014 image for a user runtime. Also, we need to expose just the 3.11 version, as users typically don‘t need a 3.8 runtime.

[the remark this answers is not in the document]
Comment (Urfoex) [reply README.md]: Moved files into root.
Adjusted base image to python:3.11.8

**Pull request merged**